File: CTask/src/org/ShapeSHifter420/Filters/AgeFilter.java
package org.ShapeSHifter420.Filters;

import org.ShapeSHifter420.Pushes.*;

public class AgeFilter implements IFilter{
    @Override
    public boolean isAllow(IPush push) {
        return push instanceof IAgePush;
    }

    @Override
    public boolean test(IPush push, ISystem sys) {
        IAgePush agePush = (IAgePush) push;
        return  (agePush.GetAge() <= sys.GetAge());
    }
}

File: CKotlin/Filters/AgeFilter.kt
package Filters

import Pushes.IAgePush
import Pushes.IPush
import Pushes.ISystemData

class AgeFilter : IFilter {
    override fun allow(push: IPush): Boolean = push is IAgePush

    override fun filtration(push: IPush,sysData:ISystemData): Boolean {
        val agePush = push as IAgePush
        return sysData.age >= agePush.age
    }
}